src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs

```diff
@@ -488,11 +488,25 @@ public bool TryRenameContext(string sourceName, string targetName)
             return result;
         }
 
+        /// <summary>
+        /// Add the input context with the specified name.
+        /// If the context with the same tenant, subscription, accountId does not exist, add the input into context list.
+        /// If the context with the same tenant, subscription, accountId already exist, merge 2 contexes and add the merged context to the context list.
+        /// </summary>
+        /// <param name="name">The specified new name of the context.</param>
+        /// <param name="context">The new context to set as default.</param>
         public bool TrySetContext(string name, IAzureContext context)
         {
             bool result = false;
             if (Contexts != null)
             {
+                string oldName = null;
+                if (TryFindContext(context, out oldName))
+                {
+                    var oldContext = Contexts[oldName];
+                    oldContext.Update(context);
```

Comment:
`context` is from user's import. Is it possible this line will change `oldContext` in `contexts`?

Reply:
Should use deepCopy. Fixed